.pre-commit-config.yaml

```diff
@@ -0,0 +1,72 @@
+---
+
+fail_fast: true
+minimum_pre_commit_version: "2.6.0"
+
+repos:
+  - # see https://github.com/pre-commit/pre-commit-hooks
+    repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.1.0
+    hooks:
+      - id: check-added-large-files
+        name: Check for accidentally added Large Files
+
+      - id: check-case-conflict
+        name: Check for cross-OS File Name Case Conflicts
+
+      - id: check-merge-conflict
+        name: Check for Git Merge Conflicts
+
+      - id: check-vcs-permalinks
+        name: Check VCS Permalinks
+
+      - id: check-json
+        name: Validate JSON files
+
+      - id: check-yaml
+        name: Validate YAML files
+
+  - # see https://github.com/antonbabenko/pre-commit-terraform
+    repo: https://github.com/antonbabenko/pre-commit-terraform
+    rev: v1.64.0
+    hooks:
+      # see https://github.com/antonbabenko/pre-commit-terraform#terraform_fmt
+      - id: terraform_fmt
+        name: Run `terraform fmt` recursively
+        args:
+          - --args=-recursive
+
+      # see https://github.com/antonbabenko/pre-commit-terraform#terraform_validate
+      - id: terraform_validate
+        name: Run `terraform validate` against `examples/basic`
+        files: examples/basic
+
+      # see https://github.com/antonbabenko/pre-commit-terraform#terraform_docs
+      - id: terraform_docs
+        name: Render documentation for Module
+        args:
+          - "--args=--config=.terraform-docs.yml"
+
+      # see https://github.com/antonbabenko/pre-commit-terraform#terraform_docs
+      - id: terraform_docs
+        name: Render documentation for `basic` example
+        files: examples/basic
+        args:
+          - "--args=--config=.terraform-docs.yml"
```

Comment:
the effect of declaring this twice is that the pre-commit looks for a `.terraform-docs.yml` file in `examples/basic`. That is a cool idea but:
1. I dont see one so im wondering how the setup is working for you. i get an error when i try this
2. If we're going to declare a specific .terraform-docs.yaml in each examples then we should suppress #L45-L48 from overwriting the file

Reply:
This is a strange one, indeed! I've generated the file a few times and it (seemed to?) work.

I'll investigate.

Reply:
I'm unable to reproduce this as not being generated. I'll close this for now but am happy to re-open if you have more pointers @drewmullen